### PR TITLE
Remove cluster provider from schema

### LINKED ISSUE #602: Remove `provider` from cluster result
Until https://github.com/pulumi/pulumi/issues/7012 is addressed, the provider returned in the cluster result is unusable without creating resources within an apply. Users should just create a provider from the returned kubeconfig instead.

Until we can address the underlying issue, we should just remove this field and consider removing the AWS provider as well, although it is not clear if there is a comparable work around.

### PR DESCRIPTION
**Pull request by jaxxstorm**
fixes #602

### CODE REVIEW

**Comment by viveklak** [comment]
I believe we also want to nuke it from the schema which should drop it from the sdk at the least: https://github.com/pulumi/pulumi-eks/blob/master/provider/cmd/pulumi-gen-eks/main.go#L164-L166

**Comment by jaxxstorm** [comment]
@viveklak I actually think nuking it from the schema is a better way, so I went ahead and did that

**Review by stack72 — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ## Unreleased
-(None)
+- Remove cluster provider from schema
```
Comment (viveklak) [inline CHANGELOG.md:2]: Maybe we need a bit more explanation for the result of the change? Something along the lines of:
```suggestion
- Due to https://github.com/pulumi/pulumi/issues/7012 including the provider in the generated SDK never really worked. This is removed now. Note - existing uses in Nodejs should not be affected.
```

**Comment by lukehoban** [comment]
This is a breaking change - right?  Building on @viveklak's comment - probably worth adding to the CHANGELOG at least here.

**Comment by stack72** [comment]
> This is a breaking change - right? Building on @viveklak's comment - probably worth adding to the CHANGELOG at least here.

This is breaking for non-TS SDKs - but the code never actually worked in those providers which is why we chose to take it on - I will make sure we prefix this as a breaking change

**Pull request merged**